### PR TITLE
Use packages filter env var in cairo-test

### PR DESCRIPTION
**Pull request by maciektr**
commit-id:83b56a17

---

**Stack**:
- #727
- #723 ⬅

⚠️ *Part of a stack created by [spr](https://github.com/ejoffe/spr). Do not merge manually using the UI - doing so may have unexpected results.*

### CODE REVIEW

File: extensions/scarb-cairo-test/src/main.rs [hunk 1]
```diff
@@ -35,6 +35,15 @@ fn main() -> Result<()> {
 
     check_scarb_version(&metadata);
 
+    ScarbCommand::new()
+        .arg("build")
+        .arg("--test")
+        .env(
+            "SCARB_PACKAGES_FILTER",
+            args.packages_filter.build_packages_filter(&metadata)?,
+        )
```
Comment (mkaput) [inline extensions/scarb-cairo-test/src/main.rs:44]: Don't we want to inherit this env from `$SCARB` process?

File: extensions/scarb-cairo-test/tests/examples.rs [hunk 2]
```diff
@@ -8,6 +8,7 @@ use test_for_each_example::test_for_each_example;
 fn cairo_test(example: &Path) {
     Command::new(cargo_bin("scarb"))
         .arg("cairo-test")
+        .arg("--workspace")
```
Comment (mkaput) [inline extensions/scarb-cairo-test/tests/examples.rs:11]: I don't understand, what does this change?

**Review by mkaput — CHANGES_REQUESTED** [review]

File: extensions/scarb-cairo-test/tests/examples.rs (hunk 2 above)
Comment (maciektr) [reply extensions/scarb-cairo-test/tests/examples.rs:11]: The workspaces example from https://github.com/software-mansion/scarb/tree/main/examples/workspaces does define the root package, so cairo-test called in workspace root would test only a single package (the root) without passing `--workspace` flag.

File: extensions/scarb-cairo-test/src/main.rs (hunk 1 above)
Comment (maciektr) [reply extensions/scarb-cairo-test/src/main.rs:44]: Actually, I think it shouldn't? 🤔 
- `scarb test` will consume the packages filter and run tests (i.e. `scarb ciaro-test`) on each package separately. Passing  `scarb test` packages filter to `scarb cairo-test` could break this behaviour. 
- `scarb cairo-test` should be able to run workspace tests on it's own, which requires building packages chosen with packages filter through scarb build.

**Review by mkaput — APPROVED** [review]

**Review by mkaput — APPROVED** [review]

**Review by mkaput — APPROVED** [review]

**Pull request merged**